packages/webdriver/src/bidi/core.ts

```diff
@@ -54,18 +55,52 @@ export class BidiCore {
             return
         }
 
-        this.#waitForConnected = new Promise<boolean>((resolve) => {
-            this.#ws.on('open', () => {
-                log.info('Connected session to Bidi protocol')
+        log.info(`Connecting to webSocketUrl ${this.#webSocketUrl}`)
+        const parsedUrl = new URL(this.#webSocketUrl)
+        // https://github.com/webdriverio/webdriverio/issues/14039
+        const candidateUrls: string[] = [this.#webSocketUrl]
+        if (!isIP(parsedUrl.hostname)) {
+            const [ips4, ips6] = await Promise.all([
+                dns.promises.resolve4(parsedUrl.hostname),
+                dns.promises.resolve6(parsedUrl.hostname),
+            ])
+            const candidateIps = [...ips4, ...ips6]
```

Comment:
ips4 and ips6 are never used
```suggestion
            const candidateIps = await Promise.all([
                dns.promises.resolve4(parsedUrl.hostname),
                dns.promises.resolve6(parsedUrl.hostname),
            ])
```

Reply:
ips4 and ips6 are both array of strings. If the above change is made then `candidateIps` type would be `string[][]`, while we want it to be `string[]`. This would work if we could use something like  [flatten](https://www.geeksforgeeks.org/lodash-_-flatten-method/) to the result of Promise.all, although I don't see lodash being present in module dependencies.

Reply:
Luckily these primitives are now [part](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) of JavaScript
```suggestion
            const candidateIps = (await Promise.all([
                dns.promises.resolve4(parsedUrl.hostname),
                dns.promises.resolve6(parsedUrl.hostname),
            ])).flat()
```